pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -75,27 +73,23 @@ func (dsl *DaemonSetLock) Acquire(metadata interface{}, TTL time.Duration) (acqu
 	}
 }
 
-func (dsl *DaemonSetLock) Test(metadata interface{}) (holding bool, err error) {
+func (dsl *DaemonSetLock) Test(metadata interface{}) (holding bool, expired bool, err error) {
 	ds, err := dsl.client.AppsV1().DaemonSets(dsl.namespace).Get(context.TODO(), dsl.name, metav1.GetOptions{})
 	if err != nil {
-		return false, err
+		return false, false, err
 	}
 
 	valueString, exists := ds.ObjectMeta.Annotations[dsl.annotation]
 	if exists {
 		value := lockAnnotationValue{Metadata: metadata}
 		if err := json.Unmarshal([]byte(valueString), &value); err != nil {
-			return false, err
-		}
-
-		if ttlExpired(value.Created, value.TTL) {
-			return true, nil
+			return false, false, err
 		}
 
-		return value.NodeID == dsl.nodeID, nil
+		return value.NodeID == dsl.nodeID, ttlExpired(value.Created, value.TTL), nil
```

Comment:
Initially my approach was to combine https://github.com/weaveworks/kured/pull/144/files/ba38d035df175a1091c35ad7ee7478fb8465bac4#diff-8baa5bb15d6f049b3a0ee81e1592fa1aR213 with https://github.com/weaveworks/kured/pull/144/files/ba38d035df175a1091c35ad7ee7478fb8465bac4#diff-8baa5bb15d6f049b3a0ee81e1592fa1aR202 and return true in case I'm not holding, but it have expired. Would that be preferred over this?